flang/lib/Lower/FIRBuilder.cpp

```diff
@@ -166,21 +166,40 @@ struct CharacterOpsBuilderImpl {
       return fir::ReferenceType::get(getCharacterType());
     }
 
-    bool isConstant() const { return data.getType().isa<fir::SequenceType>(); }
+    bool isNotInMemory() const {
```

Comment:
Values of any non-reference type are not in memory (necessarily). I suspect this query is to decide if the values ought to be lowered to memory rather than live as a 1st-class ssa-value. Is that it?

Reply:
I renamed this `needToMaterialize`.